hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/test/java/org/apache/hadoop/mapred/TestTaskAttemptListenerImpl.java

```diff
@@ -257,12 +260,6 @@ public void testGetMapCompletionEvents() throws IOException {
         createTce(3, false, TaskAttemptCompletionEventStatus.FAILED) };
     TaskAttemptCompletionEvent[] mapEvents = { taskEvents[0], taskEvents[2] };
     Job mockJob = mock(Job.class);
-    when(mockJob.getTaskAttemptCompletionEvents(0, 100))
-      .thenReturn(taskEvents);
-    when(mockJob.getTaskAttemptCompletionEvents(0, 2))
-      .thenReturn(Arrays.copyOfRange(taskEvents, 0, 2));
-    when(mockJob.getTaskAttemptCompletionEvents(2, 100))
-      .thenReturn(Arrays.copyOfRange(taskEvents, 2, 4));
```

Comment:
Encountered with this error:

```
org.mockito.exceptions.misusing.UnnecessaryStubbingException: 
Unnecessary stubbings detected.
Clean & maintainable test code requires zero unnecessary code.
Following stubbings are unnecessary (click to navigate to relevant line of code):
  1. -> at org.apache.hadoop.mapred.TestTaskAttemptListenerImpl.lambda$testGetMapCompletionEvents$2(TestTaskAttemptListenerImpl.java:267)
  2. -> at org.apache.hadoop.mapred.TestTaskAttemptListenerImpl.lambda$testGetMapCompletionEvents$2(TestTaskAttemptListenerImpl.java:269)
  3. -> at org.apache.hadoop.mapred.TestTaskAttemptListenerImpl.lambda$testGetMapCompletionEvents$2(TestTaskAttemptListenerImpl.java:271)
```
It was a  unnecessary stubbed call that was never realized during test execution.So either I have to remove it or use use 'lenient' strictness like this:
```
    lenient().when(mockJob.getTaskAttemptCompletionEvents(0, 100))
        .thenReturn(taskEvents);
    lenient().when(mockJob.getTaskAttemptCompletionEvents(0, 2))
        .thenReturn(Arrays.copyOfRange(taskEvents, 0, 2));
    lenient().when(mockJob.getTaskAttemptCompletionEvents(2, 100))
        .thenReturn(Arrays.copyOfRange(taskEvents, 2, 4));
```
I think it is better to use 'lenient' strictness

Reply:
In this case, if these stubbings are unnecessary, I think it's better to remove it to keep the code clean. What are your thoughts on that @szilard-nemeth @susheel-gupta?

Reply:
Hi @susheel-gupta ,
What is lenient strictness? 
Can you explain what's the advantage / disadvantage of it? 
If the stub call is really unnecessary and the test works fine without it, you can remove it. 
With my original comment I Jjst wanted to make sure the patch is focused to JUnit upgrade but in this case we can make an exception if it's necessary.

Reply:
To bypass strict stubs we use lenient strictness, as in new upgrade mockito is introducing new features that nudge the framework towards “strictness”.
I don't see any advantage here in this scenario, other than keeping the code as it is.
Disadvantages are test code duplication and unnecessary test code will be there if we use this 'lenient' keyword.
Yes the test works fine without it. 
So I'm removing this lines.
```
    when(mockJob.getTaskAttemptCompletionEvents(0, 100))
      .thenReturn(taskEvents);
    when(mockJob.getTaskAttemptCompletionEvents(0, 2))
      .thenReturn(Arrays.copyOfRange(taskEvents, 0, 2));
    when(mockJob.getTaskAttemptCompletionEvents(2, 100))
      .thenReturn(Arrays.copyOfRange(taskEvents, 2, 4));
```

Reply:
sure, this can work :)

Reply:
@szilard-nemeth Yes, this will work as there is no assertion for this method getTaskAttemptCompletionEvents.
If it is required to add assertions for this method, then for now I can use lenient strictness and later in another jira I can add few assertions for this method.